digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java

```diff
@@ -252,42 +262,99 @@ public TDJobOperator runJob(Config state, String key, JobStarter starter)
 
         JobState jobState = state.get(key, JobState.class, JobState.empty());
 
-        // Generate and store domain key before starting the job
+        // 0. Generate and store domain key before starting the job
+
         Optional<String> domainKey = jobState.domainKey();
         if (!domainKey.isPresent()) {
             state.set(key, jobState.withDomainKey(UUID.randomUUID().toString()));
             throw TaskExecutionException.ofNextPolling(0, ConfigElement.copyOf(state));
         }
 
-        // Start the job
+        // 1. Start the job
+
         Optional<String> jobId = jobState.jobId();
         if (!jobId.isPresent()) {
             assert domainKey.isPresent();
-            state.set(key, jobState.withJobId(starter.startJob(this, domainKey.get())));
+
+            String newJobId;
+            try {
+                newJobId = starter.startJob(this, domainKey.get());
+            }
+            catch (TDClientException e) {
+                logger.warn("failed to start job: domainKey={}", domainKey.get(), e);
+                if (isDeterministicClientException(e)) {
+                    throw e;
+                }
+                throw errorPollingException(state, key, jobState);
+            }
+
+            // Reset error polling
+            jobState = jobState.withErrorPollIteration(Optional.absent());
```

Comment:
Builder doesn't have getters. There's a new `@Value.Modifiable` annotation we could use to generate a value class with setters, but when jackson serializing an object of that class it includes at least one unwanted field. Will stick with the immutable value class for now.